### PR TITLE
Replace OpenStruct with ProposalDetail class

### PR DESCRIPTION
**Pull request by allbecauseyoutoldmeso**
### Description of change

- Use custom `ProposalDetail` class instead of `OpenStruct` to parse `planning_application.proposal_details`.

### Story Link

https://trello.com/c/GjWdNTTK/1369-openstruct-refactoring

### CODE REVIEW

File: app/components/concerns/proposal_detailable.rb [hunk 1]
```diff
@@ -0,0 +1,21 @@
+# frozen_string_literal: true
+
+module ProposalDetailable
+  extend ActiveSupport::Concern
+
+  included do
+    def formatted_policy_refs
```
Comment (allbecauseyoutoldmeso) [inline app/components/concerns/proposal_detailable.rb:7]: I've just moved this method from somewhere else, but would love suggestions for nice ways to refactor it!

File: spec/components/proposal_details/list_component_spec.rb [hunk 2]
```diff
@@ -3,92 +3,60 @@
...
+  end
+
+  it "renders first group with correct numbered questions" do
+    group_a = page.find("h4", text: "Group a").find(:xpath, "../..")
+    expect(group_a).to have_content("1. \n  \n    Test question 1")
+    expect(group_a).to have_content("2. \n  \n    Test question 3")
+  end
 
-    it "returns numbered proposal details grouped by portal name" do
-      expect(list_component.send(:groups)).to eq(
```
Comment (allbecauseyoutoldmeso) [inline spec/components/proposal_details/list_component_spec.rb:43]: I think when I initially wrote these files I'd forgotten (I think I knew once!) about how to test components nicely. So I wrote a whole bunch of tests for private methods 😬.

File: app/models/proposal_detail.rb [hunk 3]
```diff
@@ -0,0 +1,41 @@
+# frozen_string_literal: true
+
+class ProposalDetail
```
Comment (allbecauseyoutoldmeso) [inline app/models/proposal_detail.rb:3]: There's probably more logic that can be pulled into this class over time 🙂

**Review by notapatch — APPROVED** [review]
The end

**Comment by allbecauseyoutoldmeso** [comment]
> The end

😭

**Pull request merged**